src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,280 @@
+        }
+        to {
+          transform: rotate(232deg);
+        }
+      }
+      @keyframes dialEight {
+        from {
+          transform: rotate(0);
+        }
+        to {
+          transform: rotate(260deg);
+        }
+      }
+      @keyframes dialNine {
+        from {
+          transform: rotate(0);
+        }
+        to {
+          transform: rotate(290deg);
+        }
+      }
+
+      @keyframes dialZero {
+        from {
+          transform: rotate(0);
+        }
+        to {
+          transform: rotate(320deg);
+        }
+      }
+      .dialer-anim {
+        transform-origin: center;
+      }
+    `;
+  }
+
+  private static matchDigit(
+    slots: SVGAElement,
+    digit: Digit,
+    direction: Direction = Direction.normal,
+    duration = '1000ms',
+    timingFn = 'ease'
+  ) {
+    switch (digit) {
+      case Digit.zero:
+        slots?.setAttribute('style', `animation: dialZero ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.one:
+        slots?.setAttribute('style', `animation: dialOne ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.two:
+        slots?.setAttribute('style', `animation: dialTwo ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.three:
+        slots?.setAttribute(
+          'style',
+          `animation: dialThree ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.four:
+        slots?.setAttribute('style', `animation: dialFour ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.five:
+        slots?.setAttribute('style', `animation: dialFive ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.six:
+        slots?.setAttribute('style', `animation: dialSix ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.seven:
+        slots?.setAttribute(
+          'style',
+          `animation: dialSeven ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.eight:
+        slots?.setAttribute(
+          'style',
+          `animation: dialEight ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.nine:
+        slots?.setAttribute('style', `animation: dialNine ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      default:
+        slots?.removeAttribute('style');
+    }
+  }
+
+  private static addReverseDialerAnim(slots: SVGAElement, digit: number) {
+    // When the user leaves the mouse (eq of lifting the finger) it goes back
+    RotaryDialerElement.matchDigit(slots, digit, Direction.reverse, '2000ms');
+  }
+
+  private addDialerAnim(digit: number) {
+    this.handleRotation(digit);
+  }
+
+  private removeDialerAnim() {
+    const slots = this.shadowRoot?.querySelector('#slots');
+    slots?.classList.remove('dialer-anim');
+    slots?.removeAttribute('style');
+  }
+
+  private down(digit: number) {
```

Comment:
Wouldn't it be better to use `digit: Digit` in all places to keep it consistent?

Reply:
definitely :)